src/orb-tools/orb.yml

```diff
@@ -1,9 +1,149 @@
+        type: string
+    steps:
+      - run:
+          name: "Validate whether this is a well-formed orb."
+          command: circleci orb validate << parameters.orb-path >>
+
+  pack:
+    description: >
+      *Experimental*
+      Uses the CLI to pack an orb file structure
+      into a single orb yml.
+    parameters:
+      source:
+        description: "Path to the root folder of an unpacked orb."
+        type: string
+      destination:
+        description: "Path including the filename to output the packed orb."
+        type: string
+    steps:
+      - run:
+          name: "Pack << parameters.source>>"
+          command: |
+            circleci config pack << parameters.source >> > << parameters.destination >>
+
+  persist-orb-to-workspace:
+    description: >
+      Persists an orb yml between jobs in a workflow.
+    parameters:
+      orb-path:
+        description: "Path to the orb to be persisted."
+        type: string
+      workspace-path:
+        description: >
+          Save the contents of the orb into the workspace at the
+          given path. The value should include a trailing slash.
+        type: string
+    steps:
+      - run:
+          name: Save and orb to `<< parameters.workspace-path >>`
+          command: |
+            mkdir -p << parameters.workspace-path >>
+            cp << parameters.orb-path >> << parameters.workspace-path >>
+      - persist_to_workspace:
+          name: "Persist orb to `<< parameters.workspace-path >>` because that value was passed as `parameters.workspace-path`."
+          root: .
+          paths:
+            - << parameters.workspace-path >>
+
+  increment:
+    description: >
+      Uses the CLI to increment the version of an orb in the registry.
+      If the orb does not have a version yet it starts at 0.0.0
+    parameters:
+      orb-path:
+        description: "Path to an orb."
+        type: string
+      namespace:
+        description: "Namespace the orb lives under."
+        type: string
+      orb-name:
+        description: "Name of the orb to increment.;"
+        type: string
+      segment:
+        description: "The semver segment to increment 'major' or 'minor' or 'patch'"
+        type: enum
+        enum: ["major", "minor", "patch"]
+      token-variable:
+        description: "The env var containing your token. Pass this as a literal string such as `$ORB_PUBLISHING_TOKEN`. Do not paste the actual token into your configuration."
+        type: string
+    steps:
+      - run:
+          name: >
+            Increment << parameters.segment >> segment of orb at << parameters.orb-path >>
+            NOTE: this currently assumes you are publishing to the registry at circleci.com
+          command: |
+            circleci orb publish increment \
+            << parameters.orb-path >> \
+            << parameters.namespace >>/<< parameters.orb-name >> \
+            << parameters.segment >> \
+            --token << parameters.token-variable >>
```

Comment:
We should not assume this needs to be passed because someone might set up the environment to have it in the CLI config. Perhaps make it optional and only include the flag if it's present?

Reply:
Agreed 👍